plugins/kernel_v2/config/disk.rb

```diff
@@ -127,6 +136,15 @@ def validate(machine)
                            types: DEFAULT_DISK_TYPES.join(', '))
         end
 
+        if !DEFAULT_DISK_EXT.include?(@disk_ext)
+          errors << I18n.t("vagrant.config.disk.invalid_ext", ext: @disk_ext,
+                           name: @name, exts: DEFAULT_DISK_EXT.join(', ') )
```

Comment:
Mostly just so there's some kind of validation that if an extension was picked, it's something expected so they aren't presented with a potentially cryptic provider error. But yeah, since this is the generic config, it might be a chore to keep these accurate across providers :man_shrugging: :thinking:

Reply:
Would it be possible to push the validation check off to the provider in use?

Reply:
Definitely possible to have the provider to this check, it just won't be in the config validate step. So basically they won't know until Vagrant goes to create the disk. Then they'll have to destroy/re-up once they fix the extension in their Vagrantfile. I don't think there's a good way otherwise without having different separate provider config.